### PR TITLE
Zip Search Populating School Dropdown List

### PR DESCRIPTION
**Pull request by hannahbergam**
This PR utilizes the existing zip search to pull the schools by zip code into the dropdown with the other three 'default' options: Select a School, I don't teach at a school, and enter manually.

I elected to reuse the existing search but was able to pare down the search *call significantly because we are removing location search functionality. This way we are able to minimize the risk (and time) involved with writing a new search function, while streamlining performance. 

This exists currently in debouncedSearch from [SchoolAutocompleteDropdown here.](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/SchoolAutocompleteDropdown.jsx#L49)

The back end search is fairly involved and complex, and I am assuming most of these decisions were made for a clear (though perhaps not quite well documented) reason, and have elected to advise a full rewrite as a future performance enhancement rather than a P0 requirement here. [Here is the Jira ticket](https://codedotorg.atlassian.net/browse/ACQ-1717) for that followup work. 

I also added a sort functionality, an improvement over the existing search. This unfortunately highlights our duplicate schools issue, which should have a resolution soon from our RED team.

Screenshot 2024-04-03 at 1 47 58 PM

## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-1611

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by TurnerRiley — APPROVED** [review]
Looks great! Out of curiosity, do you think it would be worth making a ticket to add a Storybook or test file for SchoolZipSearch just to have in general to check that its returning the right value and that it handles having a combo of hardset values (e.g. "clickToAdd") and queried school values well?

File: apps/src/templates/SchoolZipSearch.jsx [hunk 1]
```diff
@@ -53,6 +69,7 @@ export default function SchoolZipSearch({fieldNames}) {
             styleAsText={true}
             onClick={() => {
               setInputManually(false);
+              setSelectedSchoolNcesId('selectASchool');
```
Comment (bethanyaconnor) [inline apps/src/templates/SchoolZipSearch.jsx]: nit: could this be moved into its own constant, used both here and in `SEARCH_DEFAULTS`?

**Review by bethanyaconnor — COMMENTED** [review]
Are there any tests to add here?

**Comment by hannahbergam** [comment]
> Are there any tests to add here?

@bethanyaconnor and @TurnerRiley I added a sort functionality, updated the description, and added a test with a fetch stub that verifies the sort is working! (Also there's a new screenshot to show that off)

File: apps/test/unit/templates/SchoolZipSearchTest.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  if (a === null || b === null) return false;
+  if (a.length !== b.length) return false;
+  for (var i = 0; i < a.length; ++i) {
+    if (a[i] !== b[i]) return false;
+  }
+  return true;
+};
+
+describe('SchoolDataInputs', () => {
+  allowConsoleWarnings();
```
Comment (bethanyaconnor) [inline apps/test/unit/templates/SchoolZipSearchTest.jsx]: curious why this is needed?

**Review by bethanyaconnor — APPROVED** [review]
Test looks good! One question but otherwise lgtm!

**Pull request merged**